core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -144,12 +151,17 @@ private JSONObject readInputJson(Context context, int resourceId) {
      * @return Requested category configuration object
      * @throws AmplifyException If there is a problem in the config file
      */
-    public CategoryConfiguration forCategoryType(final CategoryType categoryType) throws AmplifyException {
+    @SuppressWarnings("WeakerAccess")
+    @NonNull
+    public CategoryConfiguration forCategoryType(@NonNull CategoryType categoryType) throws AmplifyException {
         if (categoryConfigurations.containsKey(categoryType.getConfigurationKey())) {
+            //noinspection ConstantConditions We never populate null values in this map, and key does exist (above)
             return categoryConfigurations.get(categoryType.getConfigurationKey());
         } else {
-            throw new AmplifyException("Unknown/bad category type: " + categoryType,
-                    "Be sure to use one of the supported Categories in your current version of Amplify");
+            throw new AmplifyException(
+                "Unknown/bad category type: " + categoryType,
+                "Be sure to use one of the supported Categories in your current version of Amplify"
+            );
         }
```

Comment:
```
final CategoryConfiguration config = categoryConfigurations.get(categoryType.getConfigurationKey());
if (config == null) {
    return new AmplifyException(...);
}
return config;
```

I think it just flows better and we wouldn't need to do `ConstantConditions` check

Reply:
Agreed, I will update, thanks